lib/s3routes/routesUtils.ts

```diff
@@ -277,6 +278,7 @@ function okContentHeadersResponse(
     response: http.ServerResponse,
     range: [number, number] | undefined,
     log: RequestLogger,
+    shouldWriteHttpCode: boolean = true,
```

Comment:
will this not be a breaking change in javascript?

e.g. existing code will not pass the parameter, and typescript compiler will not see the default parameter value in javascript code : so calls to this function may suddenly stop sending http code, and we need to hope that we have upgraded & tested all places where it is used.

Reply:
It's only used internally but I agree that the default value should be removed, for safety...

Reply:
alternatively, depending on how it is ysed maybe it would cleaner to rework the code to either have 2 functions for the different use case, or have one function for headers and another for the response code...

Reply:
But the rationale here is more like: we don't want to change anything, so we set to true by default, except in one well-known place (streamed data). What do you think?

Reply:
PR was updated and default was removed. I confirmed this is not used anywhere else.